src/main/java/com/minecraftabnormals/upgrade_aquatic/common/entities/jellyfish/helper/JellyfishSizeMap.java

```diff
@@ -0,0 +1,117 @@
+package com.minecraftabnormals.upgrade_aquatic.common.entities.jellyfish.helper;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class JellyfishSizeMap {
+    protected final TreeMap<Float, Integer> sizeMap;
+    protected final int weightTotal;
```

Comment:
I don't see the point of this being protected when there's already a public getter for it.
I also think `totalWeight` would be a better name.
```suggestion
    private final int totalWeight;
```